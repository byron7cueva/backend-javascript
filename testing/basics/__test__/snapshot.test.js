import { getCharacter } from '../snapshot';
import rick from '../rick.json'

describe('Es hora de las instantaneas', () => {
  test('snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Siempre fallara la instantanea', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };

    expect(user).toMatchSnapshot();
  });

  test('Excepciones en el snapshot', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Byron Cueva'
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number) // Indicandole que este tip de dato va ser numero y no importa si cambia
    });
  });
});