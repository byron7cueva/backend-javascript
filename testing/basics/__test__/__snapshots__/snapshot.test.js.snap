// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2020-09-01T23:06:55.444Z,
  "id": 14,
}
`;

exports[`Es hora de las instantaneas Tenemos una excepsion 1`] = `
Object {
  "id": Any<Number>,
  "name": "Byron Cueva",
}
`;

exports[`Es hora de las instantaneas snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;
